oat-sa/tao-core
# Feature/ADF-1782/Implement the UI for the list of translations

### Related to: https://oat-sa.atlassian.net/browse/ADF-1782

Requires: 
 - [x] https://github.com/oat-sa/tao-core-ui-fe/pull/603

Also require `extension-tao-tests` and `extension-tao-itemqti` to be with the branch `feat/HKD-6/integration`

### Summary

Implement a form component for managing the list of translations for a resource.

Also implement the API client to connect with the server.

### Details

A component is now present in `form/translation` and connection with API services is given via `services/translation`.

The component shows:
- a creation form to add a translation
- a list of existing translations

When the user create a translation, or click the edit button, an event is triggered to command the editing.

![image](https://github.com/user-attachments/assets/66bb58ea-89bf-471e-b0e6-392e3177ad28)

### How to test

- checkout the branch: `git checkout -t origin/feature/ADF-1782/translation-list-ui`
- make sure to install the np resources: `cd tao/views && npm ci`
- select an item and see the translate button
- click the translate button (note: you need the item to be ready for translation, otherwise, the component will only show a placeholder message)

The following feature flags will also be needed:
```
FEATURE_FLAG_TRANSLATION_ENABLED=true
FEATURE_FLAG_UNIQUE_NUMERIC_QTI_IDENTIFIER=true
FEATURE_FLAG_TRANSLATION_DEVELOPER_MODE=true
```
